### PR TITLE
amtool check-config

### LINKED ISSUE #333: Validate config
Is there a way to check if the config is valid? (similar to promtool with Prometheus)

Thanks.

### PR DESCRIPTION
**Pull request by iksaif**
This is similar to `promtool check-config` and allows one
to validate the alertmanager configuration (as a git presubmit for example).

`govendor fetch github.com/spf13/{cobra,pflag}` was needed to
have support for `Args`.

This fixes #333 

```
$ ./amtool check-config alertmanager.yml ./doc/examples/simple.yml 
Checking 'alertmanager.yml'  SUCCESS
Found 1 templates:   FAILED: template: foo.tmpl:3: unexpected {{end}}

Checking './doc/examples/simple.yml'  SUCCESS
Found 1 templates:   SUCCESS

Error: Failed to validate 1 file(s).
```

### CODE REVIEW

**Comment by mxinden** [comment]
@iksaif Thanks a lot for the contribution. This was requested very often.

I am not familiar with the amtool testing story (//CC @stuartnelson3), but I think we should have a basic test. @iksaif would you be able to add one for the configuration check?

**Comment by stuartnelson3** [comment]
Ah I forgot to comment! I tested this manually this morning, worked great.

A really high level smoke test should be enough if it's not too much trouble -- does it recognize a good config.yml and an bad one.

**Comment by iksaif** [comment]
Added tests

File: cli/check_config_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+func TestCheckConfig(t *testing.T) {
+	err := CheckConfig([]string{"testdata/conf.good.yml"})
+	if err != nil {
+		t.Fatalf("error: %v", err)
+	}
+
+	err = CheckConfig([]string{"testdata/conf.bad.yml"})
+	if err == nil {
+		t.Fatalf("Failed to detect missing file")
```
Comment (mxinden) [inline cli/check_config_test.go]: I am confused by the word `missing` here. How about `failed to detect invalid file`?

File: cli/check_config_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package cli
+
+import (
+	"testing"
+)
+
+func TestCheckConfig(t *testing.T) {
+	err := CheckConfig([]string{"testdata/conf.good.yml"})
+	if err != nil {
+		t.Fatalf("error: %v", err)
```
Comment (mxinden) [inline cli/check_config_test.go]: Instead of `error: ` how about `checking valid config file failed with:  `?

**Comment by stuartnelson3** [comment]
couple small wording changes on the test failure output and then we'll merge

File: cli/check_config_test.go (hunk 1 above)
Comment (iksaif) [reply cli/check_config_test.go]: Forgot to change this when I changed the test

File: cli/check_config_test.go (hunk 2 above)
Comment (iksaif) [reply cli/check_config_test.go]: done

**Comment by iksaif** [comment]
done

**Pull request merged**